src/hooks/useDatasetManager.js

```diff
@@ -74,6 +74,14 @@ export const useDatasetManager = () => {
           }
         : {}
     const response = await api.dataset.get(id || datasetId, headers)
+
+    if (response?.ok === false) {
+      setError({
+        hasError: true,
+        statusCode: response.statusCode
+      })
+    }
```

Comment:
I'm not sure we want to handle errors this way. We are conditionally setting a nested attribute to false when an error occurs. I don't see this an an improvement over setting a single atomic state variable as either a status code or null and interpreting error from there.

I think something like this would be more appropriate
```suggestion
    const { ok, statusCode } = response
    if (error !== statusCode ) {
      if (ok ===  false) setError(response.statusCode)
      if (ok && statusCode) setError(null)
    }
```

Reply:
See the commit b39738f4d083c7b99be149a307824ceda204f2b6. 

I also left `TODO` comment for `ok && error` check (see this related PR #337 for reference), thank you!

Reply:
**NOTE:** PR #337 was created based on the talk regarding the `ok` property https://github.com/AlexsLemonade/refinebio-web/pull/294#discussion_r1466558333 (and first mentioned here https://github.com/AlexsLemonade/refinebio-web/pull/294#issuecomment-1911167634). This is because it requires a small refactoring of other part of the codebase (this was back in January and was old thread so wanted to add more context), thank you David!